components/setup/BuildInstruction.tsx

```diff
@@ -69,24 +90,24 @@ const BuildInstruction: React.FC<BuildInstructionProps> = ({ selectedHosting, us
 			});
 	}, [userId])
 
+	if (!installId) {
```

Comment:
There is a loader at 105 that should be returned before this condition.
Otherwise you will show the error state even when the error hasn't occurred yet.

Reply:
I don't understand why are you making this condition chain over complicated just to save on repeating 1 line (for calling the RepoSelection component).

The previous structure is so easily readable:
1. You check for hosting
2. You check if install id is available
3. You check for the installation type & provider

Now it is so difficult to understand what is the chain of conditions.